### PR TITLE
return slice(0,-1) for glb_slices if glb_numb empty on an mpi rank

### PR DESCRIPTION
**Pull request by deckerla**

### CODE REVIEW

**Comment by deckerla** [comment]
Closes #2003

**Comment by deckerla** [comment]
`local_indices` tests in MPI seem to be turned off at the moment, so I am not sure what the best way is to have unit tests for this PR

https://github.com/devitocodes/devito/blob/6e1dcc502a36b59716378cfa5e6924fcbbdb5bd5/tests/test_mpi.py#L2336

**Comment by mloubout** [comment]
>tests in MPI seem to be turned off a

No this is the "main" to debug, i.e so that you can run `python test_mpi.py`. Pytest ignores this it runs all tests functions named `test_*`. So you can just use an extra case to the test here:

https://github.com/devitocodes/devito/blob/6e1dcc502a36b59716378cfa5e6924fcbbdb5bd5/tests/test_mpi.py#L364

**Comment by deckerla** [comment]
@mloubout thanks for the advice!  most recent push includes modified unit test covering the behavior

File: tests/test_mpi.py [hunk 1]
```diff
@@ -395,24 +395,28 @@ def test_ownership(self, shape, coords, points):
         assert list(ownership.keys()) == [grid.distributor.myrank]
 
     @pytest.mark.parallel(mode=4)
-    @pytest.mark.parametrize('coords,expected', [
-        ([(0.5, 0.5), (1.5, 2.5), (1.5, 1.5), (2.5, 1.5)], [[0.], [1.], [2.], [3.]]),
+    @pytest.mark.parametrize('coords,expected,expectedinds', [
+        ([(0.5, 0.5), (1.5, 2.5), (1.5, 1.5), (2.5, 1.5)], [[0.], [1.], [2.], [3.]], [(slice(0,1),),(slice(1,2),),(slice(2,3),),(slice(3,4),)]),
+        ([(1.5, 1.5),], [[],[],[],[0.]], [(slice(0,-1),),(slice(0,-1),),(slice(0,-1),),(slice(0,1),)])
```
Comment (mloubout) [inline tests/test_mpi.py:400]: Looks good thanks. Can you check line length? We require <92 characters per line in CI.

You can run `flakke8 tests/test_mpi.py` to check (it should be installed as a devitio dependency).

**Review by mloubout — COMMENTED** [review]
Looks good thanks.

One minor CI comment but should be goog to go

File: tests/test_mpi.py (hunk 1 above)
Comment (deckerla) [reply tests/test_mpi.py]: thanks for pointing that out! should now be resolved.  the line also needed some white space after commas :)

**Pull request merged**